packages/beacon-node/src/execution/engine/http.ts

```diff
@@ -204,58 +207,44 @@ export class ExecutionEngineHttp implements IExecutionEngine {
     parentBlockRoot?: Root,
     executionRequests?: ExecutionRequests
   ): Promise<ExecutePayloadResponse> {
-    const method =
-      ForkSeq[fork] >= ForkSeq.electra
-        ? "engine_newPayloadV4"
-        : ForkSeq[fork] >= ForkSeq.deneb
-          ? "engine_newPayloadV3"
-          : ForkSeq[fork] >= ForkSeq.capella
-            ? "engine_newPayloadV2"
-            : "engine_newPayloadV1";
-
-    const serializedExecutionPayload = serializeExecutionPayload(fork, executionPayload);
+    const method = newPayloadMethodByFork(fork);
 
     let engineRequest: EngineRequest;
-    if (ForkSeq[fork] >= ForkSeq.deneb) {
-      if (versionedHashes === undefined) {
-        throw Error(`versionedHashes required in notifyNewPayload for fork=${fork}`);
-      }
-      if (parentBlockRoot === undefined) {
-        throw Error(`parentBlockRoot required in notifyNewPayload for fork=${fork}`);
-      }
-
-      const serializedVersionedHashes = serializeVersionedHashes(versionedHashes);
-      const parentBeaconBlockRoot = serializeBeaconBlockRoot(parentBlockRoot);
 
-      if (ForkSeq[fork] >= ForkSeq.electra) {
-        if (executionRequests === undefined) {
-          throw Error(`executionRequests required in notifyNewPayload for fork=${fork}`);
-        }
-        const serializedExecutionRequests = serializeExecutionRequests(executionRequests);
+    switch (method) {
+      case "engine_newPayloadV1":
+      case "engine_newPayloadV2":
         engineRequest = {
-          method: "engine_newPayloadV4",
-          params: [
-            serializedExecutionPayload,
-            serializedVersionedHashes,
-            parentBeaconBlockRoot,
-            serializedExecutionRequests,
-          ],
+          method,
+          params: generateSerializedNewPayloadParams(fork, method, {executionPayload}),
```

Comment:
while this looks much cleaner here, we lose a lot of type safety

eg. it's possible to call it like this without complaints
```ts
params: generateSerializedNewPayloadParams(fork, method, {}),
```
and then you will just notice this as runtime